server/constructors.go

```diff
@@ -11,13 +11,14 @@ import (
 	tmtypes "github.com/tendermint/tendermint/types"
 	dbm "github.com/tendermint/tm-db"
 
+	"github.com/cosmos/cosmos-sdk/store/snapshot"
 	sdk "github.com/cosmos/cosmos-sdk/types"
 )
 
 type (
 	// AppCreator is a function that allows us to lazily initialize an
 	// application using various configurations.
-	AppCreator func(log.Logger, dbm.DB, io.Writer) abci.Application
+	AppCreator func(log.Logger, dbm.DB, *snapshot.Store, io.Writer) abci.Application
```

Comment:
The snapshot store along with the policy will be passed as options to the baseapp in the app's constructor (correct me if I'm wrong). Why not just have the constructor itself create the snapshot store if it wishes instead of modying the `AppCreator`?

Reply:
I'm cleaning this up a bit now. I would really like to just default to a `snapshots` directory under `BaseConfig.RootDir`. The only place where we have access to `RootDir` is e.g. here: https://github.com/cosmos/cosmos-sdk/blob/6eb2ffed55010f65d303c0106e0e26271d0b3ec1/server/start.go#L180

Do you have any suggestions for how to pass this to the application automatically without modifying `AppCreator`?

Reply:
Nevermind, decided to handle this in Gaia instead, placing snapshots under `gaiad/snapshots/`.